### PR TITLE
Bump to airflow 2.6.2

### PR DESCRIPTION
**Pull request by rsevilla87**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [x] Optimization
- [ ] Documentation Update

## Description

Bumping airflow version from 2.3.2 to 2.6.2. Includes some nice features and bug fixes affecting us like:
- Renewed GUI
- TaskGroups are marked as green when all child tasks are successful
- Task log streaming for kubernetes tasks. (no more page refresh \o/ )

Also:
- Airflow playgrounds have now edge termination by default
- Removing pip statement to install airflow, elasticsearch and prometheus-cli modules. Installing these in the Dockerfile modules is problematic since it can upgrade the core airflow module due to dependency requirements.
  - I've verified that these airflow modules are included in the base image (docker.io/apache/airflow:2.6.2-python3.8) by default:
  ```console
  $ podman run --rm --entrypoint bash -it apache/airflow:2.6.2-python3.8
  $ pip list | egrep "apache-airflow-providers-(cncf-kubernetes|elastic)"
  apache-airflow-providers-cncf-kubernetes 7.0.0
  apache-airflow-providers-elasticsearch   4.5.0
  ```
  - elasticsearch is also available by default
  - prometheus-cli is not used anymore

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Comment by Harshith-umesh** [comment]
even the nocp manifest needs to be updated: https://github.com/cloud-bulldozer/airflow-kubernetes/blob/81a0039b47478aa20501511932ccf59c4c86cb1b/dags/nocp/manifest.yaml#L10

File: dags/common/models/dag_config.py [hunk 1]
```diff
@@ -20,6 +20,6 @@ class DagConfig:
         })
     executor_image: Optional[dict] = field(default_factory=lambda: {
             "repository": "quay.io/cloud-bulldozer",
-            "tag": "2.3.2"
+            "tag": "2.6.2"
```
[the remark this answers is not in the document]
Comment (rsevilla87) [reply dags/common/models/dag_config.py:23]: well, this variable actually overwritten but I'll change it anyway

[the remark this answers is not in the document]
Comment (rsevilla87) [reply dags/common/models/dag_config.py:23]: I ended up updating the tag by `latest` so we don't get confused again.

File: images/airflow-ansible/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-ARG BASE_IMAGE=quay.io/cloud-bulldozer/airflow:2.3.2
+ARG BASE_IMAGE=quay.io/cloud-bulldozer/airflow:latest
```
Comment (jtaleric) [inline images/airflow-ansible/Dockerfile:1]: I realize this is our own repo, but could tracking latest be a bad idea?

Comment (rsevilla87) [reply images/airflow-ansible/Dockerfile:1]: this variable is actually overwritten by podman, so we can ignore it

Comment (rsevilla87) [reply images/airflow-ansible/Dockerfile]: Just removed the default value to avoid confusion

**Comment by rsevilla87** [comment]
I have to figure out what's wrong with the dependencies when running the tests...

File: images/airflow-managed-services/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-ARG BASE_IMAGE=quay.io/cloud-bulldozer/airflow:2.3.2
+ARG BASE_IMAGE=quay.io/cloud-bulldozer/airflow:latest
 # Hypershift Compilation
 FROM golang:1.18 AS hypershift
```
Comment (mukrishn) [inline images/airflow-managed-services/Dockerfile:3]: Hypershift CLI needs latest go, so please make it `1.19` along with this PR.

**Review by mukrishn — APPROVED** [review]
LGTM, I've tested HCP builds.

**Pull request merged**